documentation/docs/25-build-and-deploy/60-adapter-cloudflare.md

```diff
@@ -107,6 +107,70 @@ Cloudflare Workers specific values in the `platform` property are emulated durin
 
 For testing the build, you should use [Wrangler](https://developers.cloudflare.com/workers/wrangler/) **version 3**. Once you have built your site, run `wrangler pages dev .svelte-kit/cloudflare`.
 
+## Pages Middleware
+
+You can deploy one middleware function that closely follows the [Pages Middleware API](https://developers.cloudflare.com/pages/functions/middleware/). You can use it to intercept requests even for prerendered pages. Combined with using [server-side route resolution](configuration#router) you can make sure it runs prior to all navigations, no matter client- or server-side. This allows you to for example run A/B-tests on prerendered pages by rerouting a user to either variant A or B depending on a cookie.
+
+> [!NOTE] It isn't really Pages Middleware because the adapter compiles to a [single `_worker.js` file](https://developers.cloudflare.com/pages/platform/functions/#advanced-mode) (also see the [Notes](#Notes) section), which ignores middleware, but it closely mirrors its capabilities.
+
+To get started, place a `cloudflare-middleware.js` file in your `src` folder and export a `onRequest` function from it:
+
+```js
+/// file: cloudflare-middleware.js
+// @filename: ambient.d.ts
+declare module '@cloudflare/workers-types';
+
+// @filename: index.js
+// ---cut---
+import { normalizeUrl } from '@sveltejs/kit';
+
+/**
+ * @param {import('@cloudflare/workers-types'.EventContext)} context
```

Comment:
there's a typo here...

```suggestion
 * @param {import('@cloudflare/workers-types').EventContext} context
```

...but with the fix, I get this error:

> Generic type 'EventContext' requires 3 type argument(s).

Do we need to expose our own interface here?